python-sdk/conftest.py

```diff
@@ -248,25 +249,26 @@ def remote_files_fixture(request):
     for item_count in range(file_count):
         object_prefix = f"test/{unique_value}{item_count}.{file_extension}"
         bucket_name, hook, object_path = _upload_or_delete_remote_file(
-            file_create, object_prefix, provider, source_path
+            file_create, object_prefix, provider, source_path, conn_id
         )
         object_path_list.append(object_path)
         object_prefix_list.append(object_prefix)
 
     yield object_path_list
 
-    if provider == "google":
-        for object_prefix in object_prefix_list:
+    for object_prefix in object_prefix_list:
+        if provider == "google":
             # if an object doesn't exist, GCSHook.delete fails:
             hook.exists(bucket_name, object_prefix) and hook.delete(  # skipcq: PYL-W0106
                 bucket_name, object_prefix
             )
-    elif provider == "amazon":
-        for object_prefix in object_prefix_list:
+        elif provider == "amazon":
             hook.delete_objects(bucket_name, object_prefix)
+        elif provider == "azure":
+            hook.delete_file(bucket_name, object_prefix)
 
 
-def _upload_or_delete_remote_file(file_create, object_prefix, provider, source_path):
+def _upload_or_delete_remote_file(file_create, object_prefix, provider, source_path, conn_id):  # noqa: C901
```

Comment:
Are you overloading this method? How does it handle the other file location using this function with `conn_id` added?

Reply:
@sunank200 at the moment, the fixture uses the default connections for AWS and GCP

- https://github.com/astronomer/astro-sdk/blob/main/python-sdk/conftest.py#L280
- https://github.com/astronomer/astro-sdk/blob/main/python-sdk/conftest.py#L293

Would you rather we use the azure default connection?

And it may be good to add support to this fixture pass the conn_id to any provider as well

Reply:
we going to change to use the default connection for azure, and we will remove this logic.

`export AIRFLOW_CONN_WASB_DEFAULT='wasb://blob%20username:blob%20password@myblob.com?tenant_id=tenant+id'
`
https://airflow.apache.org/docs/apache-airflow-providers-microsoft-azure/stable/connections/wasb.html

Reply:
Yes whenever we have azure default connection, we can use that.

Reply:
https://github.com/astronomer/astro-sdk/issues/1446